### PR TITLE
[video_player] Fix memory leak issue

### PR DESCRIPTION
**Pull request by xiaowei-guan**
OnVideoFrameDecoded callback may be received during destory MM player. So check is_initialized_ value when receive frame decoded callback.

### CODE REVIEW

**Review by wanchao-xu — APPROVED** [review]
Looks good to me

File: packages/video_player/tizen/src/video_player.cc [hunk 1]
```diff
@@ -158,7 +158,19 @@ VideoPlayer::VideoPlayer(flutter::PluginRegistrar *plugin_registrar,
   SetUpEventChannel(plugin_registrar->messenger());
 }
 
-VideoPlayer::~VideoPlayer() { Dispose(); }
+VideoPlayer::~VideoPlayer() {
```
Comment (swift-kim) [inline packages/video_player/tizen/src/video_player.cc:161]: Is it no longer necessary to call `Dispose` from the destructor? Should a similar thing be applied to video_player_videohole for consistency?

Comment (xiaowei-guan) [reply packages/video_player/tizen/src/video_player.cc:161]:
```cpp
std::optional<FlutterError> VideoPlayerTizenPlugin::Dispose(
    const TextureMessage &msg) {
  LOG_DEBUG("[VideoPlayerTizenPlugin] textureId: %ld", msg.texture_id());

  auto iter = players_.find(msg.texture_id());
  if (iter != players_.end()) {
    iter->second->Dispose();
    players_.erase(iter);
  }

  return std::nullopt;
}
```
we don't need call dispose int destructor,  will cause call dispose() twice.

**Comment by haesik** [comment]
Can this PR solve the robot video player issue?

File: packages/video_player/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 2.4.6
+
+* Fix memory leak issue.
+
 ## NEXT
 
 * Increase the minimum Flutter version to 3.3.
```
Comment (swift-kim) [inline packages/video_player/CHANGELOG.md:7]:
```suggestion
## 2.4.6

* Fix a memory leak issue.
* Increase the minimum Flutter version to 3.3.
```

Comment (xiaowei-guan) [reply packages/video_player/CHANGELOG.md]: ok

**Review by swift-kim — APPROVED** [review]

**Comment by swift-kim** [comment]
Was @haesik's question answered?

**Comment by xuelian-bai** [comment]
memory leak issue has been solve

**Pull request merged**